api/api_public_transaction_pool.go

```diff
@@ -149,6 +150,48 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
 	return nil
 }
 
+// GetDecodedAnchoringTransactionByHash returns the decoded anchoring data of anchoring transaction for the given hash
+func (s *PublicTransactionPoolAPI) GetDecodedAnchoringTransactionByHash(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	var tx *types.Transaction
+
+	// Try to return an already finalized transaction
+	if tx, _, _, _ = s.b.ChainDB().ReadTxAndLookupInfo(hash); tx != nil {
+		goto decode
+	}
+
+	// No finalized transaction, try to retrieve it from the pool
+	if tx = s.b.GetPoolTransaction(hash); tx != nil {
+		goto decode
```

Comment:
Why don't you change if conditions to `tx == nil` and then `return nil, errors`?
In that case, we can remove goto statement which may help to make spaghetti code later

Reply:
I aligned this code with `GetTransactionByHash`. 
There are 2 cases that can be decoded. So If this if statement is changed as you said, we can't remove `goto`.
If we add 1 more case that can be decoded, I think this code is better.